[FEAT] 대학교 리스트 화면 UI

tea-hkim opened the pull request:
## 작업내용
<!-- 작업 내용과 이미지를 첨부해주세요. -->
- [x] 대학교 리스트 화면 UI 작업
- [x] 컴포넌트 shadowTextFieldView 추가
- [x]  대학교 이름 검색 기능 추가

<!-- (+스크린샷)이 있다면 적어주세요. 없으면 지워주세요-->
|대학교 리스트 - 검색 전|대학교 리스트 - 검색 후|대학교 리스트 - 공백 포함 검색 후|
|:---:|:---:|:---:|
|<img width="250" src="https://user-images.githubusercontent.com/81206228/196221460-c61a1474-0e8b-4c93-8841-e754b3307e79.png">|<img width="250" src="https://user-images.githubusercontent.com/81206228/196221469-8f5f2ff9-0534-439c-bd55-14d025800a03.png">|<img width="250" src="https://user-images.githubusercontent.com/81206228/196232680-69b27d73-33b0-4069-9eb9-3ed1607ba547.png">|


### 1.  대학교 리스트 화면 UI 작업 
- 화면의 타이틀을 26이라는 고정값을 주었는데 디바이스의 글씨 크기에 따라 유동적으로 변하지 않아서 방법을 찾고 있습니다.
    나중에 다시 PR올리겠습니다.
- system 내부에 있는 color를 사용했습니다. 이 부분도 추후에 변경해서 디자인 변경 PR 올리겠습니다
- shadow를 가진 TextField는 추후에 사용될 가능성이 있어보여 컴포넌트로 만들었습니다.

### 2. 대학교 이름 검색 기능 e68529e
스페이스가 들어오는 상황을 예외 처리하기 위한 코드입니다.

1. 입력 값이 없다면 기본 대학 리스트를 띄운다.
2. 입력 값이 있고, 입력 값에 공백이 있다면
     1. 스페이스를 기준으로 하는 문자열 배열을 만든다.
     2. 문자열 내부에 있는 문자를 가진 대학교 이름들을 필터링 한다.
     3. 필터링 된 새로운 리스트를 띄운다.
4. 입력 값이 있고, 입력 값에 공백이 없다면 그대로 필터링을 한다.
    1. 필터링 된 새로운 리스트를 띄운다.

- 이렇게 작업을 한 이유는 유저가 띄어쓰기를 통해 2가지 이상의 단어를 검색하는 이유가 
   단어들의 교집합을 통해 필터링을 하려는 것이 아니라 3가지 단어를 전부 포함하고 있는 합집합을 찾으려고 한다 생각했기 때문입니다.
- 이후 생기는 예외 처리 상황들을 찾는다면 PR에 추가하도록 하겠습니다.

## 다음 작업 내용
- [ ] 대학교 이메일 입력 화면 UI 
- [ ] 대학교 리스트 cell 탭 시 대학교 이메일 입력 화면으로 전환

## 리뷰포인트
<!-- 리뷰가 필요한 포인트와 해당 되는 커밋을 링크로 걸어주세요. -->
1. 대학교 이름 검색 기능 e68529e 을 만들었는데 아주 지저분한 코드가 되었습니다. 
    더 좋은 방법을 남겨주신다면 열심히 공부하겠습니다. (알고리즘 공부의 중요성을 느꼈습니다...)
3. 공부한 MVVM 패턴에선 지금의 ViewModel에서 로직을 분리해내어 UseCase를 만들어야 한다고 생각하는데
    ViewModel의 크기를 더 줄이는게 좋을까요??

## Checklist
- [x] 브랜치를 가져와 작업한 경우 이전 브랜치에 PR을 보냈는지 확인
- [x] 빌드를 위해 SceneDelegate 수정한 것 PR로 올리지 않았는지 확인
- [x] 필요없는 주석, 프린트문 제거했는지 확인
- [x] 컨벤션 지켰는지 확인
- [x] final, private 제대로 넣었는지 확인

tea-hkim committed:
Feat:sparkles:: 대학교 검색 뷰 UI 저장

대학교 검색 뷰의 OrgSearchingViewController와 ViewModel을 만들었습니다
Combine을 이용하여 연결을 시도하는 상태이고 다음 커밋 때 연결하여 명시할 예정입니다

tea-hkim committed:
Feat:sparkles:: OrgListCell 추가

OrgSearchingViewController의 tableView의 cell을 추가하였습니다.

tea-hkim committed:
Feat:sparkles:: 대학교 검색 기능 추가

combine을 이용해 목데이터를 받아와 화면에 띄우는 기능을 구현하였습니다.

tea-hkim committed:
Feat:sparkles:: TextField의 왼쪽 이미지 붙이는 기능 추가

tea-hkim committed:
Chore:broom:: OrgListCell 왼쪽 패딩값 변경

tea-hkim committed:
Chore:broom:: UIOrganizationPlaceHolder 삭제

tea-hkim committed:
Feat:sparkles:: OrgListCell에 disclosureIndicator 추가

tea-hkim committed:
Feat:sparkles:: ShadowTextFieldView 추가

텍스트 필드를 위한 ShadowTextFieldView를 만들었습니다.
 - 작업 전, textField에 leftView로 추가했던 imageView를 ShadowTextFieldView의
   내부에 추가했습니다.
 - ShadowTextFieldView의 내부에 textField 프로퍼티를 만들어 활용할 수 있도록 하였습니다

tea-hkim committed:
Fix:beetle:: 스페이스를 검색하면 결과가 아나오는 버그 수정

검색 기능을 수정하였습니다.
  - 기존 검색에서는 스페이스를 넣어서 검색하면 검색이 되지 않는 현상이 있었습니다
  - 조건문을 통해 검색 기능의 예외 처리를 했습니다
  - 조건문 말고 다른 좋은 방법이 있다면 알려주시면 감사하겠습니다

tea-hkim committed:
Chore:broom:: TODO 정리

TODO와 swiftLint에서 주의를 주는 부분을 정리하였습니다

tea-hkim committed:
Chore:broom:: final, private 설정

tea-hkim committed:
Fix:beetle:: 공백이 들어오면 검색 결과가 나오지 않는 현상 해결

tea-hkim committed:
Chore:broom:: 불필요한 함수 제거

tea-hkim committed:
Chore:broom:: 검색 기능에서 반복되는 코드 삭제

tea-hkim committed:
Chore:broom:: 파일 이름 변경

파일들 이름이 통일되지 않아서 헷갈릴 상황이 발생할 우려가 생겼습니다.
  - 기존의 'OrgSearching'이라는 키워드를 'OrganizatinoList'로 변경했습니다
  - 'Org'라는 약어를 'Organization'으로 변경했습니다

tea-hkim committed:
Chore:broom:: MockData 기존 버전으로 변경

공백이 들어간 대학교 이름을 검색하기 위해 목데이터를 변경했습니다.
변경한 부분을 되돌려 놨습니다

tea-hkim committed:
Chore:broom:: DesignSystem/Asset conflict 해결

tea-hkim committed:
Merge branch 'dev' into feat/OrgSetting/ZES-100-searchingOrgList

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+//
+//  OrganizationListViewController.swift
```
p2
ViewController를 View 폴더안에 위치하도록 해주세요

tea-hkim replied:
130dc01 반영했습니다!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+import UIKit
+import Combine
+import SnapKit
+import DesignSystem
```
p3
퍼스트파티 -> 서드파티
알파벳 순으로 정렬해주세요

tea-hkim replied:
e3e76df 반영했습니다!

Gobans replied:
Design system도 서드파티 취급하기로 해서.. UIKit 밑에 놓아주세요!

tea-hkim replied:
eb677f8 반영했습니다!

Gobans replied:
snapkit 서드파티입니다..! 알파벳순으로 정렬 해주세용🥲

tea-hkim replied:
0e723b7 수정했습니다!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+    
+    // MARK: Properties
+    
+    private var viewModel = OrganizationListViewModel()
```
p2
var -> let viewModel은 클래스입니다!

tea-hkim replied:
e3e76df 반영했습니다!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+        }
+    }
+    
+    private func bindUI() {
```
p2
bindUI 를 extension으로 빼서 클래스 바로 밑에 위치로 옮겨주세요

tea-hkim replied:
130dc01 반영했습니다!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+
+extension OrganizationListViewController {
+    
+    // MARK: UI Function
```
p2
UI Function은 가장 아래에 위치해도 된다고 생각합니다. (bindUI -> delegate 등 기능의 중요도 순으로 배치하는게 좋을 것 같아요)

tea-hkim replied:
ffd6d17 변경했습니다!
bindUI라는 mark 외에도 delegate라는 mark를 추가했습니다!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: OrganizationListCell.identifier, for: indexPath) as? OrganizationListCell
+        guard let cell = cell else { return UITableViewCell()}
```
p3
한칸 띄워주세요!
```suggestion
        guard let cell = cell else { return UITableViewCell() }
```

tea-hkim replied:
130dc01 반영했습니다!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+    }
+    
+    private func bindUI() {
+        self.viewModel.$searchingArray.sink { orgNameArray in
```
p1
weak self 필요합니다!
```suggestion
        self.viewModel.$searchingArray
        .sink { [weak self ] orgNameArray in
        guard let self = self else { return }
```

Gobans replied:
++ viewModel의 searchingArray -> view의 orgNameArray 로 데이터를 가져오면 안될 것 같아요

tea-hkim replied:
130dc01 반영했습니다! 현재는 빌드되어 돌아가도록 해놨는데 구현 부분에 있어서는 좀 더 생각을 해보고 리팩토링 하겠습니다!

tea-hkim replied:
555462d 변경하였습니다! 모든 로직이 ViewModel에서 돌아가도록 하였습니다

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+        self.viewModel.$searchingArray.sink { orgNameArray in
+            self.orgNameArray = orgNameArray
+            self.tableView.reloadData()
+        }.store(in: &subscriptionSet)
```
p3
가독성을 위해 바꾸면 좋을 것 같아요
```suggestion
        }
        .store(in: &subscriptionSet)
```

tea-hkim replied:
130dc01 반영했습니다

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+        }.store(in: &subscriptionSet)
+        
+        self.searchingTextField
+            .userInputTextPublisher
```
p2
extension에 UITextField의 publisher를 정의해놓았습니다. 바꿔주세요!

tea-hkim replied:
5493c11 수정했습니다!

Gobans replied:
p4
```swift
searchingTextField.textDidEndEditingPublisher
                                .map
                                ...
                                ...
```


 로  publisher 밑에 명령어들로 라인을 주는게 가독성이 좋을 것 같아요

tea-hkim replied:
6b786e1 해결하였습니다! 확실히 가독성이 좋아졌네요

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+            self.tableView.reloadData()
+        }.store(in: &subscriptionSet)
+        
+        self.searchingTextField
```
p3
안써도 되는 self는 모두 지워주세요 (많은 swift convention guideline에서 필요없는 self는 지양하고 있습니다)

tea-hkim replied:
130dc01 반영했습니다!
VeiwController 내부의 self는 수정했는데 나머지 파일은 다른 commit에서 수정하도록 하겠습니다.

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+
+extension OrganizationListViewController: UITextFieldDelegate {
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        self.view.endEditing(true)
```
p3
self 삭제  부탁드립니다

tea-hkim replied:
130dc01 반영했습니다!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+                                             object: self)
+        .compactMap { $0.object as? UITextField }
+        .compactMap { $0.text }
+        .eraseToAnyPublisher()
```
p2
extension에 UITextField의 publisher를 정의해놓았습니다. 바꿔주세요!

tea-hkim replied:
5493c11 수정했습니다!
제가 extension의 존재를 잊고있었네요...

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/View/Cell/OrganizationListCell.swift:
```diff
@@ -0,0 +1,63 @@
+    // MARK: UI Function
+    
+    private func configureUI() {
+        self.addSubviews([orgNameLabel, disclosureIndicator])
```
p3
이 파일안에 있는 self 들도 모두 교체해주세요!

tea-hkim replied:
aec256c 수정했습니다!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/View/Cell/OrganizationListCell.swift:
```diff
@@ -0,0 +1,63 @@
+        orgNameLabel.snp.makeConstraints { make in
+            make.left.equalTo(self.snp.left).inset(20)
+            make.top.equalTo(self.snp.top).inset(16)
+            make.bottom.equalTo(self.snp.bottom).inset(16)
```
p4
에버리가 알려준 팁
```suggestion
            make.top.bottom.equalToSuperview().inset(16)
```

tea-hkim replied:
664db90 수정했습니다!
찾아보니 top과 bottom 모두 적용되는 `verticalEdges`라는 키워드가 있어서 이걸 사용하여 다음과 같이 변경하였습니다

```Swift

disclosureIndicator.snp.makeConstraints { make in
      make.right.equalTo(snp.right).inset(20)
      make.verticalEdges.equalToSuperview().inset(16)
}
```

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/ViewModel/OrganizatinoListViewModel.swift:
```diff
@@ -0,0 +1,63 @@
+        didSet {
+            searchingInput(userTextInput)
+        }
+    }
```
p3
property observer를 combine으로 구현해보면 어떨까요?

tea-hkim replied:
555462d 변경하였습니다!!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/ViewModel/OrganizatinoListViewModel.swift:
```diff
@@ -0,0 +1,63 @@
+    
+}
+
+extension OrganizationListViewModel {
```
[Q]
extension으로 나눈 이유는 무엇인가요?

tea-hkim replied:
메서드 역할을 하는 것들을 나눠서 보기 쉽게 하려는 의도입니다!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+    
+    private var subscriptionSet = Set<AnyCancellable>()
+    
+    private var orgNameArray: [String] = []
```
p1
ViewController가 가지고 있을 필요가 없어보입니다. View 는 viewModel이 가진 데이터에 따라 화면을 그려주는 역할만 하도록 해주세요

tea-hkim replied:
130dc01 변경했습니다!!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/ViewModel/OrganizatinoListViewModel.swift:
```diff
@@ -0,0 +1,63 @@
+    }
+    
+    // Output
+    @Published var searchingArray: [String] = []
```
p2
searchedOrgArray 가 더 의미상 자연스러운 것 같아요

Lia316 replied:
Org 넣는 거 찬성이요! 
Array는 취향 차이라서,, 저는 List를 더 선호하긴 합니다 😗➿🎶

tea-hkim replied:
2073086 수정했습니다!
- searchedOrgArray로 변경했습니다

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/ViewModel/OrganizatinoListViewModel.swift:
```diff
@@ -0,0 +1,63 @@
+    
+    // Input
+    private var organizationArray: [Organization] = []
+    private var orgNameArray: [String] = []
```
p3
Input output 구분은 Publisher에 해당할 때 적어주는게 좋을 것 같아요

Lia316 replied:
> Input output 구분은 Publisher에 해당할 때 적어주는게 좋을 것 같아요

고반! 이건 어떤 의미실까요? 👀

저는 뷰에게 받아서 가공해야할 정보는 input,
뷰에게 최종적으로 전달할 가공된 정보는 output이라고 생각하고 있습니다

tea-hkim replied:
130dc01 변경했습니다!
- Publisher에 해당하는 것들은 따로 분리하여 input oupt 구분을 하였습니다
- @Published 키워드가 붙지 않은 프로퍼티들은 최상단으로 올렸습니다

Gobans replied:
@Lia316 닉이 수정하신대로 프로퍼티와 publisher의 분리를 말씀 드린 것 입니다!!

Gobans commented on Projects/DesignSystem/Sources/Component/ShadowTextFieldView.swift:
```diff
@@ -0,0 +1,81 @@
+        textField.textColor = .black
+        textField.font = UIFont.boldSystemFont(ofSize: 16)
+        
+        leftImage.image = UIImage(.magnifyingglass)
```
Q
SF symbol 말고 UIImage asset을 넣어 만든 이유가 무엇인가요?

tea-hkim replied:
SF symbol과 디자인에서 사용된 돋보기의 모양이 달라서 디자인에서 사용된 이미지를 사용하였습니다!

Chaeho-Min replied:
음...?! 저 SF Symbol 사용했습니다!

tea-hkim replied:
아 폰트 굵기가 달랐네요... 기호로 전환할 수 있는 방법을 찾아 commit 하겠습니다

Gobans replied:
혹시 이 부분은 해결이 되었나요?

tea-hkim replied:
해결 중에 있습니다 :) 해결하면 바로 commit 올리겠습니다

tea-hkim replied:
555462d 변경하였습니다! 필요없는 이미지들은 삭제하고 system Image로 변경하였습니다

Gobans commented on Projects/DesignSystem/Sources/Component/ShadowTextFieldView.swift:
```diff
@@ -0,0 +1,81 @@
+    }
+    
+    private func createLayout() {
+        // TODO: addSubViews extension으로 교체하기
```
지금 교체해주세요!

tea-hkim replied:
6b786e1 교체하였습니다!

Gobans commented on Projects/DesignSystem/Sources/Component/ShadowTextFieldView.swift:
```diff
@@ -0,0 +1,81 @@
+public final class ShadowTextFieldView: UIView {
+    public let textField = UITextField()
+    private let leftImage = UIImageView()
+    private let backgroundView = UIView()
```
p2
backgroundView가 딱히 필요없는 view 같습니다. 해당 뷰의 appearance를 ShadowTextFieldView (self)가 하면 안될까요?

tea-hkim replied:
ShadowTextFieldView는 도화지가 되는 View라고 생각합니다. 
그 위에 총 3개의 레이어가 필요하다고 생각했습니다.
1. leftImage, textField
2. backgoundView (하얀 배경의 width를 가지고 있는)
3. shadowView (그림자 역할을 하는)

이렇게 3개의 레이어를 둔 이유는 최상단이 textField만 있다면 메인 Appearance가 될 수 있겠지만,
같은 층에 leftImage를 가지고 있기 때문에 이 두 가지 컴포넌트의 배경이 될 View가 필요하다고 생각합니다.

Gobans replied:
그렇군요 상세한 설명 감사합니다!

chaneeii commented on Projects/DesignSystem/Sources/Asset/Asset.swift:
```diff
@@ -9,6 +9,7 @@
 import Foundation
 
 public enum Asset: String {
+    case magnifyingglass
```
p2)
컨벤션 통일을 위함입니당
ic_magnifyingglass

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/ViewModel/OrganizatinoListViewModel.swift:
```diff
@@ -0,0 +1,63 @@
+        searchingArray = orgNameArray
+    }
+    
+    private func searchingInput(_ input: String) {
```
p3
searchInput 이 더 적절한 것 같습니다

tea-hkim replied:
130dc01 변경했습니다!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/ViewModel/OrganizatinoListViewModel.swift:
```diff
@@ -0,0 +1,63 @@
+        orgNameArray = organizationArray.map { $0.name }
+    }
+    
+    private func setSearchingData() {
```
p3
setInitialSearchedArray() 로 변수명을 짓고 아래
```swift
   if input.isEmpty {
             setInitialSearchedArray()
             return
         }
```
이렇게 써주는게 가독성이 더 좋을 것 같아요

tea-hkim replied:
130dc01 변경했습니다!

chaneeii commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+    
+    private var viewModel = OrganizationListViewModel()
+    
+    private var subscriptionSet = Set<AnyCancellable>()
```
p1)
저희 요부분 다같이 네이밍 통일하는게 좋을 것 같습니다..!

@tea-hkim @Lia316 @Gobans @italsehf @Chaeho-Min

Gobans replied:
cancelBag 한표!

Lia316 replied:
cancelBag 두 표요

tea-hkim replied:
e3e76df 변경했습니다!

chaneeii replied:
리액터킷 만든,, 한국 rx 장인이신 전수열님이 요런것도 만들어두셨더라구요..! 안쓰겠지만 있다고 공유하고싶었어요!
https://github.com/devxoul/CancelBag

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/ViewModel/OrganizatinoListViewModel.swift:
```diff
@@ -0,0 +1,63 @@
+final class OrganizationListViewModel {
+    
+    // Input
+    private var organizationArray: [Organization] = []
```
p2
org 나 organization 줄이거나 다 쓰거나 통일 하면 좋을 것 같아요

tea-hkim replied:
130dc01
약어를 쓰기보다 명시를 해주는 것이 나중을 위해서도 좋을 것 같아서 org -> organization으로 변경했습니다!

Lia316 replied:
통일하자는 의견 좋아요!!

변수명은 org 찬성, 모델명은 길게 그대로 가는 것 주장합니다!

Gobans replied:
이거 아직 수정이 안된 것 같아요🥲

tea-hkim replied:
dfdd6ea 수정하였습니다 변수명은 org로 통일했습니다!

chaneeii commented on Projects/App/Sources/UI/Organization/OrganizationList/View/Cell/OrganizationListCell.swift:
```diff
@@ -0,0 +1,63 @@
+    
+    private func createLayout() {
+        orgNameLabel.snp.makeConstraints { make in
+            make.left.equalTo(self.snp.left).inset(20)
```
p5)
.equalToSuperview() 를 사용하면 더 편리할것 같습니다! 추천!

Gobans commented on Projects/DesignSystem/Sources/Component/ShadowTextFieldView.swift:
```diff
@@ -0,0 +1,81 @@
+    }
+}
+
+extension ShadowTextFieldView {
```
p3
extension 위 아래 한칸 띄워주세요

Gobans commented on Projects/DesignSystem/Sources/Component/ShadowTextFieldView.swift:
```diff
@@ -0,0 +1,81 @@
+            make.width.equalToSuperview()
+            make.height.equalToSuperview()
+        }
+    }
```
p3
extension 위 아래 한칸 띄워주세요

chaneeii reviewed (approved):
👍 고생하셨습니다

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/View/Cell/OrganizationListCell.swift:
```diff
@@ -0,0 +1,63 @@
+            make.right.equalTo(self.snp.right).offset(-20)
+            make.bottom.equalTo(self.snp.bottom).offset(-16)
+        }
+    }
```
p3
extension 위 아래 한칸 띄워주세요

tea-hkim replied:
664db90 변경하였습니다!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+    }
+}
+
+extension OrganizationListViewController: UITextFieldDelegate {
```
p3
extension 위 아래 한칸 띄워주세요

tea-hkim replied:
130dc01 반영했습니다!

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/OrganizationListViewController.swift:
```diff
@@ -0,0 +1,124 @@
+            .receive(on: DispatchQueue.main)
+            .assign(to: \.userTextInput, on: viewModel)
+            .store(in: &subscriptionSet)
+    }
```
p3
한칸 띄워주세요

tea-hkim replied:
130dc01 반영했습니다!

Gobans reviewed (changes requested):
고생하셨습니다 닉!
저도 그랬지만 템플릿에 맞춰 띄어쓰기, extension 분리를 다시 검토해보실 필요가 있을 것 같아요!
코드 리뷰 내용을 간략하게 남겼는데, 리뷰에 대한 자세한 설명이나 의견은 언제든지 말해주시면 좋을 것 같아요.
파이팅~!

tea-hkim committed:
Chore:broom:: OrganizatinoListViewModel request 반영

  - 'org'키워드 'organization'으로 변경 : 아무래도 명시하는 게 좋을 것 같아서 약어를 사용하지 않기로 했습니다!
  - input/ouput @published에 해당하는 것들로만 구성
  - 프로퍼티, 메서드 이름 변경

tea-hkim committed:
Merge Origin branch to local branch

tea-hkim committed:
Chore:broom:: OrganizationListViewController request 적용

  - cancelBag 변경
  - ViewController 내부의 데이터 ViewModel로 이동
  - bindUI 위치 변경
  - extension 위 아래 여백 추가

tea-hkim committed:
Chore:broom:: Delegate extension 위치 변경

tea-hkim committed:
Chore:broom:: searchedList 이름 변경

searchedList를 searchedOrgList로 변경했습니다

tea-hkim committed:
Chore:broom:: textFieldPublisher 추가

기존에 만들어놨던 publiser를 없애고, extension에 있는 textFieldPublisher를 추가하였습니다

tea-hkim committed:
Chore:broom:: cell 내부의 self 키워드 제거

tea-hkim committed:
Chore:broom:: tableView separateStyle 변경

tea-hkim committed:
Chore:broom:: snpakit 메서드 변경

  - 기존에 right과 left를 나눠서 사용한던 것을 horizantalEdges
  - top과 bottom을 verticalEdges로 변경하였습니다

tea-hkim committed:
Chore:broom:: import 순서 변경

DesignSystem을 최하단으로 내렷습니다!

tea-hkim committed:
Chore:broom:: TODO violation 해결

 - ShadowTextFieldView의 addViews를 해결하였습니다
 - 코드의 가독성을 높이기 위해 점 접근법을 제대로 세팅하였습니다

tea-hkim committed:
Chore:broom:: Organization 변수명 Org로 통일

tea-hkim committed:
Chore:broom:: 디펜던시 관련 수정

tea-hkim committed:
Chore:broom:: 검색 기능 ViewModel로 이동

기존에 ViewController에서 ViewModel의 searchedArray 받기 위한 빈 배열을 삭제하고
ViewModel에서 searchedArray를 직접 받아올 수 있도록 변경하였습니다!
 - 중간에 index error가 발생했는데 이는 tableView의 reload가 의도한 시점에 일어나지 않아서 발생하는 현상이었습니다.
 - tableView의 reload를 DispatchQueue의 main 쓰레드에서 일어나도록 하였습니다 (UI의 랜더링은 main 쓰레드에서 해야 한다는 기초를 까먹어서...)

Gobans reviewed (approved):
수고하셨습니다!!! 👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍
